pkg/virt-controller/watch/vmi_test.go

```diff
@@ -3021,6 +3024,100 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			Expect(controller.cidsMap.reverse).To(BeEmpty())
 		})
 	})
+
+	Context("dynamic interface attachment", func() {
+		const vmName = "vm1"
+
+		var (
+			pod *k8sv1.Pod
+			vmi *virtv1.VirtualMachineInstance
+		)
+
+		expectPodStatusUpdateFailed := func(pod *k8sv1.Pod) {
+			kubeClient.Fake.PrependReactor("patch", "pods", func(action testing.Action) (handled bool, obj k8sruntime.Object, err error) {
+				_, ok := action.(testing.PatchAction)
+				Expect(ok).To(BeTrue())
+				return true, pod, fmt.Errorf("your cluster is doomed")
+			})
+		}
+
+		fakeHotPlugRequest := func(vmi *virtv1.VirtualMachineInstance, addOpts []virtv1.AddInterfaceOptions) {
+			for _, req := range addOpts {
+				vmi.Spec.Networks = append(
+					vmi.Spec.Networks,
+					virtv1.Network{
+						Name: req.InterfaceName,
+						NetworkSource: virtv1.NetworkSource{
+							Multus: &virtv1.MultusNetwork{NetworkName: req.NetworkName},
+						},
+					})
+			}
+		}
+
+		const (
+			firstVMInterface  = "oldIface1"
+			firstVMNetwork    = "oldnet1"
+			secondVMInterface = "oldiface2"
+			secondVMNetwork   = "oldnet2"
+		)
+
+		Context("k8s API is down - i.e. you cannot update the pod status", func() {
```

Comment:
Over indenting makes it very hard to read the tests.
If there is no strong need, I would avoid it as much as possible.

It looks like a lot of lines and indentation could have been avoided by making this a simple `It` test.

Reply:
If I make a single `It` I do not have access to this particular test setup - I think having a clear separation between the test setup and the test itself is valuable enough to introduce the context.

The fact the identation is pushed to the right by the additional context if offset the the new-lines I usually add to make sure the code line length stays small, for instance:
```golang
				vmi = newVMIWithOneNetwork(
					newVMIWithOneNetwork(
						api.NewMinimalVMI(vmName), firstVMNetwork, firstVMInterface),
					secondVMNetwork, secondVMInterface)
```

instead of:
```golang
vmi = newVMIWithOneNetwork(newVMIWithOneNetwork(api.NewMinimalVMI(vmName), firstVMNetwork, firstVMInterface), secondVMNetwork, secondVMInterface)
```